drivers/misc/renesas_rx_dtc/renesas_rx_dtc.c

```diff
@@ -0,0 +1,330 @@
+	const struct dtc_renesas_rx_config *config = dev->config;
+
+	if (!is_valid_activation_irq(activation_irq)) {
+		return -EINVAL;
+	}
+
+	/* Re configuration */
+	if (!data->dtc_vt_status[activation_irq]) {
+		dtc_renesas_rx_disable_transfer(activation_irq);
+		dtc_renesas_rx_wait_for_transfer(dev, activation_irq);
+	}
+
+	rx_dtc_block_repeat_initialize(p_info);
+
+	/* Disable read skip prior to modifying settings. */
+	config->reg->DTCCR.BIT.RRS = 0;
+
+	/* Update the entry in the DTC Vector table. */
+	gp_dtc_vector_table[activation_irq] = p_info;
+
+	/* Enable read skip after all settings are written. */
+	config->reg->DTCCR.BIT.RRS = 1;
+
+	data->dtc_vt_status[activation_irq] = DTC_ACT_CONFIGURED;
+	return 0;
+}
+
+int dtc_renesas_rx_start_transfer(const struct device *dev, uint8_t activation_irq)
+{
+	struct dtc_renesas_rx_data *data = dev->data;
+	int ret;
+
+	if (!is_valid_activation_irq(activation_irq)) {
+		return -EINVAL;
+	}
+
+	if (data->dtc_vt_status[activation_irq] == DTC_ACT_IDLE) {
+		return -EACCES;
+	}
+
+	ret = dtc_renesas_rx_enable_transfer(activation_irq);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->dtc_vt_status[activation_irq] = DTC_ACT_IN_PROGRESS;
+
+	return 0;
+}
+
+int dtc_renesas_rx_stop_transfer(const struct device *dev, uint8_t activation_irq)
+{
+	struct dtc_renesas_rx_data *data = dev->data;
+	int ret;
+
+	if (!is_valid_activation_irq(activation_irq)) {
+		return -EINVAL;
+	}
+
+	if (!data->dtc_vt_status[activation_irq]) {
+		return -EACCES;
+	}
+
+	ret = dtc_renesas_rx_disable_transfer(activation_irq);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Clear pointer in vector table. */
+	data->dtc_vt_status[activation_irq] = DTC_ACT_IDLE;
+	gp_dtc_vector_table[activation_irq] = NULL;
+	return 0;
+}
+
+int dtc_renesas_rx_reset_transfer(const struct device *dev, uint8_t activation_irq,
+				  void const *p_src, void *p_dest, uint16_t const num_transfers)
+{
+	const struct dtc_renesas_rx_config *config = dev->config;
+	transfer_info_t *const gp_dtc_vector = gp_dtc_vector_table[activation_irq];
+	int err = 0;
+
+	if (!is_valid_activation_irq(activation_irq)) {
+		return -EINVAL;
+	}
+
+	err = dtc_renesas_rx_disable_transfer(activation_irq);
+	if (err < 0) {
+		return err;
+	}
+
+	dtc_renesas_rx_wait_for_transfer(dev, activation_irq);
+
+	config->reg->DTCCR.BIT.RRS = 0;
+	/* Reset transfer based on input parameters. */
+	if (NULL != p_src) {
+		gp_dtc_vector->p_src = p_src;
+	}
+
+	if (NULL != p_dest) {
+		gp_dtc_vector->p_dest = p_dest;
+	}
+
+	if (TRANSFER_MODE_BLOCK == gp_dtc_vector->transfer_settings_word_b.mode) {
+		gp_dtc_vector->num_blocks = num_transfers;
+	} else if (TRANSFER_MODE_NORMAL == gp_dtc_vector->transfer_settings_word_b.mode) {
+		gp_dtc_vector->length = num_transfers;
+	} else {
+		/* Do nothing */
+	}
+	config->reg->DTCCR.BIT.RRS = 1;
+	dtc_renesas_rx_enable_transfer(activation_irq);
+
+	return err;
+}
+
+int dtc_renesas_rx_infoGet(const struct device *dev, uint8_t activation_irq,
```

Comment:
@quytranpzz , please use snake case

```suggestion
int dtc_renesas_rx_info_get(const struct device *dev, uint8_t activation_irq,
```